release notes/v0.4.0.md

```diff
@@ -0,0 +1,171 @@
+- `locator.waitFor([options])` ([#406](https://github.com/grafana/xk6-browser/pull/406))
+
+
+## Bugs fixed
+
+- Fixed zombie browser processes. ([#306](https://github.com/grafana/xk6-browser/pull/306))
+
+  Before this fix, browser processes remained active even after the test or k6 test runs. This could sometimes saturate the machine and therefore could affect the tests.
+
+- Fixed cleanup of the browser's temporary directory. ([#323](https://github.com/grafana/xk6-browser/pull/323), [#409](https://github.com/grafana/xk6-browser/pull/409))
+
+  The temporary directory is now properly removed when `Browser.close()` is called, or when the iteration or test run completes. This will help ensure that any private data that the browser caches in the temporary directory is deleted.
+
+  NOTE: There is a small chance that the temporary directory will not be cleaned and a ticket has been opened to track the next step/fix.  ([#403](https://github.com/grafana/xk6-browser/issues/403))
+
+- Fixed keyboard press of modifier keys (e.g. `Shift`) in `keyboard.down` and `keyboard.up`. ([#325](https://github.com/grafana/xk6-browser/pull/325))
+
+- Fixed `page.content()` so that it no longer panics. ([#327](https://github.com/grafana/xk6-browser/pull/327))
+
+- Fixed `ElementHandle.fill()`. ([#383](https://github.com/grafana/xk6-browser/pull/383))
+
+  Before, it was not filling in the text in the selected input element.
+
+- Fixed `ElementHandle.tap()`. ([#401](https://github.com/grafana/xk6-browser/pull/401))
+
+  Before, it was performing the tap without respecting the vertical coordinate.
+
+- Fixed timeout detection. ([#411](https://github.com/grafana/xk6-browser/pull/411))
+
+  Before this fix, we couldn't detect the timeouts properly. For example if we write a test where we are retrieving the contents of an element that doesn't exist, it used to return an empty string, but now it will raise a timeout error correctly. The current system is still unstable because this fix re-surfaced a data race bug that happens from time to time (happens on frame navigation). Previously reported in [#254](https://github.com/grafana/xk6-browser/issues/254)
+
+
+## Improvements
+
+- Permissions, such as microphone and camera, can now be granted to the browser.  ([#314](https://github.com/grafana/xk6-browser/pull/314))
+
+  This grants the new browser window permissions that would allow tests to use those devices. You can test the change using the example in [examples/grant_permission.js](https://github.com/grafana/xk6-browser/blob/v0.4.0/examples/grant_permission.js)
+
+- We're also removing the slow-motion option for `Frame.isChecked()` since it doesn't make sense to slow down this sort of check ([#376](https://github.com/grafana/xk6-browser/pull/376))
+
+- We're working on making xk6-browser available in the k6 Cloud! The feature is still in early testing stages, but we hope to make more announcements about it soon. Stay tuned!
```

Comment:
```suggestion
- Our log messages are now more concise and user friendlier. ([#429](https://github.com/grafana/xk6-browser/pull/429))

- We're working on making xk6-browser available in the k6 Cloud! The feature is still in early testing stages, but we hope to make more announcements about it soon. Stay tuned!
```

Reply:
🤔 

```suggestion
- The error messages are now more concise and user-friendlier. ([#429](https://github.com/grafana/xk6-browser/pull/429))

- We're working on making xk6-browser available in the k6 Cloud! The feature is still in the early testing stages, but we hope to make more announcements about it soon. Stay tuned!
```

Reply:
@ankur22 I think it should be "user-friendlier". User-friendliness is a relative concept, but we made an improvement.

I'm not sure I agree with all of your changes @inancgumus, but maybe we can agree on this:

> Some of our log messages are now more concise and user-friendlier.

I don't think we can say that all of them are, but it's an improvement. So "some of", and "our" instead of "the", as we're not changing e.g. the site logs. And "log messages" instead of "error messages", since there might be some warnings changed there as well; the change wasn't focused on error messages, even though those are the majority.

Reply:
I agree with "some of" or "most"? Here's my suggestion, but I don't have a strong opinion:

> Most of the log messages are now more concise and user-friendlier. We're still working on improving the rest to make them easy to figure out.